chat.js

```diff
@@ -920,25 +925,38 @@ Chat.uncacheTree = function (root) {
 	} while (uncache.length > 0);
 };
 
-Chat.loadCommands = function () {
-	if (Chat.commands) return;
+Chat.loadPlugins = function () {
+	if (Chat.commands || Chat.filter) return;
 
 	FS('package.json').readTextIfExists().then(data => {
 		if (data) Chat.package = JSON.parse(data);
 	});
 
 	let baseCommands = Chat.baseCommands = require('./chat-commands').commands;
 	let commands = Chat.commands = Object.assign({}, baseCommands);
+	let chatfilters = [];
 
-	// Install plug-in commands
+	// Install plug-in commands and chat filter
 
 	// info always goes first so other plugins can shadow it
 	Object.assign(commands, require('./chat-plugins/info').commands);
 
 	for (let file of FS('chat-plugins/').readdirSync()) {
 		if (file.substr(-3) !== '.js' || file === 'info.js') continue;
 		Object.assign(commands, require('./chat-plugins/' + file).commands);
+
+		const filter = require(`./chat-plugins/${file}`).chatfilter;
+		if (filter) chatfilters.push(filter);
 	}
+
+	Chat.filter = function (message, user, room, connection, targetUser) {
+		if (!chatfilters.length) return message;
+
+		for (let i = 0; i < chatfilters.length; i++) {
+			const output = chatfilters[i].call(this, message, user, room, connection, targetUser);
+			if (typeof output === 'string') return message;
```

Comment:
I'm still confused what you mean with this context.

If a filter returns an altered version of the message, then that is what their message would become.  All "chat loggers" would then log the altered message, not the one that they had tried to send.  That's how it currently works, so why change that?